### PR TITLE
Cookies string splitter

### LINKED ISSUE #13: Incompatibility with Fetch API
The cookie parser does not recognize response returned by the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API). That makes it nearly impossible to use the library in browser/Node.js/React Native environments.

The library assumes response to be a plain object with the `set-cookie` property. However, the [Fetch Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) has a different interface:
```js
response.headers.get('Set-Cookie')
```
```js
response.headers.getAll('Set-Cookie')
```

### LINKED ISSUE #18: Split cookies string
I made PR: https://github.com/joeferraro/react-native-cookies/pull/97
which parses multiple comma separated cookies in one string.

Discussion here as well:
https://github.com/nfriedly/set-cookie-parser/issues/13

Probably this functionality should go here instead?
This method returns array of cookie strings which this library can parse.

### PR DESCRIPTION
**Pull request by chrusart**
Set-Cookie header returned by few frameworks is a one string of comma separated cookie strings.
This PR adds method to split this string into array of separate cookie strings.
Handles commas in parameters like expires.

e.g. string:
```
"cid=70125eaa-399a-41b2-b235-8a5092042dba; expires=Thu, 04-Jun-2020 12:17:56 GMT; Max-Age=63072000; Path=/, client_id=70125eaa-399a-41b2-b235-8a5092042dba; Max-Age=63072000; Path=/; expires=Thu, 04-Jun-2020 12:17:56 GMT"
```
to array:
```
[
"cid=70125eaa-399a-41b2-b235-8a5092042dba; expires=Thu, 04-Jun-2020 12:17:56 GMT; Max-Age=63072000; Path=/",
"client_id=70125eaa-399a-41b2-b235-8a5092042dba; Max-Age=63072000; Path=/; expires=Thu, 04-Jun-2020 12:17:56 GMT",
]
```

Fixes #18 
Fixes #13

### CODE REVIEW

**Comment by nfriedly** [comment]
@tomball: This PR is a JavaScript implementation of your code at https://github.com/google/j2objc/commit/16820fdbc8f76ca0c33472810ce0cb03d20efe25 

1. Thank you for your code!
2. I think we're in the clear since this is a new implementation, but I'd still prefer it if we had your blessing on releasing this code as MIT, (because your original code is Apache 2.0). If absolutely necessary, I'd be willing to change the licensing on this library to Apache 2.0, but I'd prefer to leave it as-is.
3. If you have the time, I appreciate an extra pair of eyes reviewing the code for bugs.

**Comment by nfriedly** [comment]
Oh, and I'm considering having this behavior on by default; can anyone think of any scenarios where that would break something that currently works properly?

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 Accepts a single `set-cookie` header value, an array of `set-cookie` header values, or a Node.js response object that may have 0 or more `set-cookie` headers.
 
-Also accepts an optional options object. Defaults: 
+Also accepts an optional options object. Defaults:
```
Comment (tomball) [inline README.md]: Unnecessary change.

Comment (tomball) [inline README.md]: Unnecessary trailing space.

File: README.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ Always returns an array of cookie objects. Each object will have, at a minimum a
 * domain - domain for the cookie (string or undefined, may begin with "." to indicate the named domain or any subdomain of it)
 * expires - absolute expiration date for the cookie (Date object or undefined)
 * maxAge - relative max age of the cookie in seconds from when the client receives it (integer or undefined)
- * Note: when using with [express's res.cookie() method](http://expressjs.com/en/4x/api.html#res.cookie), multiply `maxAge` by 1000 to convert to miliseconds. 
+ * Note: when using with [express's res.cookie() method](http://expressjs.com/en/4x/api.html#res.cookie), multiply `maxAge` by 1000 to convert to miliseconds.
```
Comment (tomball) [inline README.md]: Ditto (consider turning on "strip trailing whitespace" in your text editor).

File: test/index.js [hunk 3]
```diff
@@ -73,3 +73,180 @@ describe('set-cookie-parser', function () {
...
+  });
+
+  it('should return empty array on non string type', function () {
+    var actual = splitCookiesString(1);
+    var expected = [];
+    assert.deepEqual(actual, expected);
+  });
+
+  it('should parse empty string', function () {
+    var actual = splitCookiesString(1);
```
[the remark this answers is not in the document]
Comment (tomball) [reply test/index.js]: LGTM

[the remark this answers is not in the document]
Comment (chrusart) [reply test/index.js]: Ahh, tests passes cause it's not a string and return empty array anyway, honestly, it should be ```' '``` there.

[the remark this answers is not in the document]
Comment (chrusart) [reply test/index.js]: I'm not really get what you mean about json with commas @nfriedly , this code parsing one single string and handles commas in cookie values already.
Test passes @tomball cause it's not a string and then empty array is returned, but it should be an empty string there.

**Comment by chrusart** [comment]
I guess it could be used in parse by default.
But the logic is up to you @nfriedly .

1. I guess ```input.headers['set-cookie']``` is a node.js thing and can have multiple ```'Set-Cookie'``` header fields, returns array of single cookies (array even if one right?, not sure, never used) and they for sure cannot be multiple cookies strings (right?), so we have:
```
if (node.js) {
   // it's clear, we have array of single cookie strings
} else { 
  // string or array of strings
  // can this array have a string that contains multiple cookies???
}

2. ```else```, I don't think it's gonna be a performance issue if we use splitting method on string every time. If we have array of strings, can it have multiple cookies in single string, if yes (I don't know) then we have to parse every array string value with split method and merge single cookie strings all toghether.

I don't think It could break something because we just split cookies from string, logic when to do it in parse is the only thing.

File: test/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (nfriedly) [reply test/index.js]: I was thinking of a new test for the JSON value, unrelated this one. I'll write it up and show you what I meant.

**Comment by nfriedly** [comment]
It's uncommon, but not impossible that Node.js would receive multiple set-cookies concatenated into a single header field-value. 


So I think making it the default behavior everywhere is sensible as long as it doesn't break any other scenarios. I'll make it a major version bump when I do it, though. This will land in 2.x as a non-default option.

**Comment by nfriedly** [comment]
This is what I meant for a JSON cookie test: https://github.com/nfriedly/set-cookie-parser/commit/a689b7f530996f48e78149378c541a9bd37f3f2e

**Comment by nfriedly** [comment]
This should be released in v2.2.0 once CI passes

**Comment by chrusart** [comment]
Ahhh, didn't think that there could be a json string.
Just curious, when this could happen? Or it's some extra functionality of this library to parse it?

**Comment by nfriedly** [comment]
> Just curious, when this could happen?

People can put any value they want into a cookie, and I've definitely seen client-side code that sets JSON cookies, so I wouldn't be surprised to see server-side code do it somewhere.

> Or it's some extra functionality of this library to parse it?

There's nothing built-into this library to parse JSON, I just wanted to ensure that we wouldn't break it.

**Comment by chrusart** [comment]
Ok, thanks!
